File: EASY/Contador de palabras/codigo.rb
# Contador de palabras en Ruby

def contar_palabras(texto)
    palabras = texto.split(/\W+/).reject { |palabra| palabra.empty? }
    total_palabras = palabras.length
    total_caracteres = texto.length
    palabras_unicas = palabras.uniq.length
  
    puts "Número total de palabras: #{total_palabras}"
    puts "Número total de caracteres: #{total_caracteres}"
    puts "Número de palabras únicas: #{palabras_unicas}"
  end
  
  puts "Introduce el texto para contar las palabras:"
  texto = gets.chomp
  contar_palabras(texto)
  

File: EASY/Contador de palabras/codigo.py
# Contador de palabras en Python

def contar_palabras(texto):
    palabras = texto.split()
    total_palabras = len(palabras)
    total_caracteres = len(texto)
    palabras_unicas = len(set(palabras))

    print(f"Número total de palabras: {total_palabras}")
    print(f"Número total de caracteres: {total_caracteres}")
    print(f"Número de palabras únicas: {palabras_unicas}")

if __name__ == "__main__":
    texto = input("Introduce el texto para contar las palabras: ")
    contar_palabras(texto)
